### DISCUSSION TITLE
Multiplayer Problem

### DISCUSSION THREAD

**Discussion by Nomadhellhound** [mupen64plus-ae/mupen64plus-ae#934] [Q&A]
Just bought the game for android and when I try to get a code to play online,  I receive a message saying "Unable to get code" on every game. I need help to know what is going on, love the app but bought the Pro version just for the online multiplayer...

**Comment by fzurita** [comment]
Worse comes to worse, I can issue a refund on the pro version.

Here is all you need for getting a code to work correctly:
* The server has to be connected to a wifi router
* The wifi router that the server is using must support UPnP.
* The UPnP service in your router must provide the exact ports that the app asks for. 
* Your ISP must allow running a server.

If any of those fail, you will not be able to get a code.

If you can't get it to work, please send me an email with your personal email address and I can issue a refund using that.

**Reply by hardsaul12** [reply]
Hola compre el juego . Tengo todo en orden  lo que mencionas y sigue sin conectar . Como puedo pedir un reembolso de mi dinero? Te dejo mi correo . Hard.saul.12@gmail.com  . Espero tu respuesta !  Gracias

**Comment by Nomadhellhound** [comment]
I checked the port forwarding and the upnp settings and all looks ok, but
the app is unable to give me a code anyways. Love your work and It is
unfortunate that it won't work for me. This is my email and I believe it is
the same I use on the Play Store. ***@***.***

Em qua., 11 de ago. de 2021 20:04, Francisco Zurita <
***@***.***> escreveu:

**Reply by fzurita** [reply]
Unfortunately, I can't see your email here. Please find my email in my github profile and email me there with it.

**Comment by etm64-cpu** [comment]
No me genera el código , no sé por qué 😞 lo acabo de comprar hace horas y simplemente no se genera, me podrían dar una solución?

**Comment by jeldermiranda02-coder** [comment]
Hola! Me puedes dar un reembolso, ya que la app pro que descargue, no funciona a como debe ser, he seguido todos los pasos a como debe ser, y aún así no funciona el multijugador, por favor espero tu respuesta gracias.

**Reply by jeldermiranda02-coder** [reply]
Este es mi correo jeldermiranda02@gmail.com
